neqo-transport/src/cc/classic_cc.rs

```diff
@@ -300,6 +310,13 @@ impl<T: WindowAdjustment> CongestionControl for ClassicCongestionControl<T> {
         if !pkt.cc_in_flight() {
             return;
         }
+        if !self.app_limited() {
+            // Given the current non-app-limited condition, we're fully utilizing the congestion
+            // window. We can safely consider that all in-flight packets up to the current to be not
+            // app-limiting. However, the subsequent packets might fall under app-limited
+            // condition. Hence, set the `first_app_limited` to the next packet number.
+            self.first_app_limited = pkt.pn + 1;
```

Comment:
Maybe you should just track the last non-app-limited packet instead.

Reply:
Hmm, I shouldn't do code review during meetings at 3am.  That's what you had before.

I don't understand why this would be different than what you had before.

Reply:
Before packet 0 was always non-app-limited so increased the cwnd by 1MSS (if not lost). Some tests checked if only the first packet was send+received, that the cwnd stayed the same. To keep these tests running, I switched to this approach. It feels a little bit like a workaround, but I would be ok with it.

Reply:
Oh, I see what is going on now.  That's unfortunate.  I would prefer that we increase CWND for the first packet.  We should increase CWND if we get an ACK; skipping the first packet for that makes no real sense.  Especially when you consider that for the entirety of slow start, that missed packet will double its effect in terms of lost opportunity to send each round trip.

If the scope of changes here is too large, then maybe we can make the change separately, but I think that we should change what tests expect from the CWND.

Reply:
The 0 packet still does count towards cwnd if we utilize the half of the cwnd during the first RTT. It only doesn't count towards cwnd if we sent out 5 or less packets. That might be fine?